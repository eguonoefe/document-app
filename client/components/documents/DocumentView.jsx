import React from 'react';
import { Modal } from 'react-materialize';
import ReactHtmlParser from 'react-html-parser';
import TinyMCE from 'react-tinymce';
import PropTypes from 'prop-types';

import InputField from '../common/InputField';
import Dropdown from '../common/Dropdown';

/**
 * Document view component for viewing a document
 * Has update and delete modals for deleting and updating a document
 *
 * @param {Object} props { id, title, content, access, onSubmit,
 * onChange, getContent, deleteDocument, userId, authorId }
 * @returns {String} HTML markup for the Document view
 */
const DocumentView = (props) => {
  const {
    id,
    title,
    content,
    access,
    onSubmit,
    onChange,
    getContent,
    deleteDocument,
    userId,
    authorId } = props;
  return (
    <div className="card blue-grey darken-1">
      <div className="card-content white-text">
        <span className="card-title">{title}</span>
        <div>{ReactHtmlParser(content)}</div>
        <p>{access}</p>
      </div>
      <div className="card-action">
        <div className="row">
          <div className="col l6 m6 s12">
            {userId === authorId && <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal" id="delete-document">
                  Delete
                </a>}
              actions={
                <div>
                  <button
                    className="btn btn-flat modal-action modal-close"
                    name={id} onClick={deleteDocument}
                    id="delete">
                    Delete
                </button>
                  <button
                    className="btn btn-flat modal-action modal-close left">
                    Cancel
                  </button>
                </div>}>
              <div >
                <h5
                  className="center">
                  Are you sure you want to delete the user
                </h5>
              </div>
            </Modal>
            }
          </div>
          <div className="col l6 m6 s12">
            {userId === authorId && <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal"
                  id="update-document"
                >
                  Update
                </a>
              }
              actions={
                <div>
                  <button
                    id="save-update"
                    className="modal-action modal-close btn btn-flat"
                    onClick={onSubmit}>
                    Update
                      </button>
                  <button
                    className="btn btn-flat modal-action modal-close left">
                    Cancel
                      </button>
                </div>
              }>
              <div >
                <h5>Update Document</h5>
                <InputField
                  name="title"
                  value={title}
                  placeholder="Document Title"
                  className="validate form-design"
                  type="text" onChange={onChange} />
                <TinyMCE
                  content={content}
                  config={{
                    plugins: 'link image code',
                    height: 200,
                    toolbar:
                    `undo redo |
                    bold italic | alignleft aligncenter alignright | code`
                  }}
                  onChange={getContent}
                />
                <select
                  name="access"
                  value={access}
                  className="browser-default input-field select"
                  onChange={onChange}>
                  <Dropdown value="" text="Select access type" />
                  <Dropdown value="public" text="Public" />
                  <Dropdown value="private" text="Private" />
                  <Dropdown value="role" text="Role" />
                </select>
              </div>
            </Modal>}
          </div>
        </div>
      </div>
    </div>
  );
};

DocumentView.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  getContent: PropTypes.func,
  deleteDocument: PropTypes.func.isRequired,
  id: PropTypes.number,
  userId: PropTypes.number,
  authorId: PropTypes.number,
  title: PropTypes.string,
  content: PropTypes.string,
  access: PropTypes.string
};

export default DocumentView;
