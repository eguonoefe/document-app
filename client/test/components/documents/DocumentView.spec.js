/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import DocumentView from '../../../components/documents/DocumentView';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/documents/DocumentView');

describe('DocumentView Component', () => {
  test('should match the document view snapshot', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const deleteDocument = jest.fn();
    const component = shallow(
      <DocumentView
        title={'Book'}
        id={3}
        content={'Awesome'}
        access={'public'}
        onChange={onChange}
        onSubmit={onSubmit}
        deleteDocument={deleteDocument}
        userId={3}
        authorId={3} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
