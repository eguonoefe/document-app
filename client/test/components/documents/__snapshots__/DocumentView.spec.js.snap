// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentView Component should match the document view snapshot 1`] = `
<div
  className="card blue-grey darken-1"
>
  <div
    className="card-content white-text"
  >
    <span
      className="card-title"
    >
      Book
    </span>
    <div>
      Awesome
    </div>
    <p>
      public
    </p>
  </div>
  <div
    className="card-action"
  >
    <div
      className="row"
    >
      <div
        className="col l6 m6 s12"
      >
        <Modal
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                id="delete"
                name={3}
                onClick={[MockFunction]}
              >
                Delete
              </button>
              <button
                className="btn btn-flat modal-action modal-close left"
              >
                Cancel
              </button>
            </div>
          }
          bottomSheet={false}
          fixedFooter={false}
          id="modal-1"
          options={
            Object {
              "dismissible": true,
              "endingTop": "10%",
              "inDuration": 250,
              "onCloseEnd": null,
              "onCloseStart": null,
              "onOpenEnd": null,
              "onOpenStart": null,
              "opacity": 0.5,
              "outDuration": 250,
              "preventScrolling": true,
              "startingTop": "4%",
            }
          }
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="delete-document"
            >
              Delete
            </a>
          }
        >
          <div>
            <h5
              className="center"
            >
              Are you sure you want to delete the user
            </h5>
          </div>
        </Modal>
      </div>
      <div
        className="col l6 m6 s12"
      >
        <Modal
          actions={
            <div>
              <button
                className="modal-action modal-close btn btn-flat"
                id="save-update"
                onClick={[MockFunction]}
              >
                Update
              </button>
              <button
                className="btn btn-flat modal-action modal-close left"
              >
                Cancel
              </button>
            </div>
          }
          bottomSheet={false}
          fixedFooter={false}
          id="modal-1"
          options={
            Object {
              "dismissible": true,
              "endingTop": "10%",
              "inDuration": 250,
              "onCloseEnd": null,
              "onCloseStart": null,
              "onOpenEnd": null,
              "onOpenStart": null,
              "opacity": 0.5,
              "outDuration": 250,
              "preventScrolling": true,
              "startingTop": "4%",
            }
          }
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="update-document"
            >
              Update
            </a>
          }
        >
          <div>
            <h5>
              Update Document
            </h5>
            <InputField
              className="validate form-design"
              name="title"
              onChange={[MockFunction]}
              placeholder="Document Title"
              type="text"
              value="Book"
            />
            <TinyMCE
              config={
                Object {
                  "height": 200,
                  "plugins": "link image code",
                  "toolbar": "undo redo |
                    bold italic | alignleft aligncenter alignright | code",
                }
              }
              content="Awesome"
            />
            <select
              className="browser-default input-field select"
              name="access"
              onChange={[MockFunction]}
              value="public"
            >
              <Dropdown
                text="Select access type"
                value=""
              />
              <Dropdown
                text="Public"
                value="public"
              />
              <Dropdown
                text="Private"
                value="private"
              />
              <Dropdown
                text="Role"
                value="role"
              />
            </select>
          </div>
        </Modal>
      </div>
    </div>
  </div>
</div>
`;
